pkgs/native_toolchain_c/lib/src/cbuilder/cbuilder.dart

```diff
@@ -95,26 +105,63 @@ class CBuilder implements Builder {
   /// Defaults to `true` for libraries and `false` for executables.
   final bool? pic;
 
+  /// The language standard to use.
+  ///
+  /// When set to `null`, the default behavior of the compiler will be used.
+  final String? std;
+
+  /// Whether to compile sources as C++.
+  ///
+  /// [cppLinkStdLib] only has an effect when this is `true`.
+  final bool cpp;
+
+  /// The C++ standard library to link against.
+  ///
+  /// This option has no effect when [cpp] is `false` or when compiling for
+  /// Windows.
+  ///
+  /// When set to `null`, the following defaults will be used, based on the
+  /// target OS:
+  ///
+  /// | OS      | Library      |
+  /// | :------ | :----------- |
+  /// | Android | `c++_shared` |
+  /// | iOS     | `c++`        |
+  /// | Linux   | `stdc++`     |
+  /// | macOS   | `c++`        |
+  /// | Fuchsia | `c++`        |
+  final String? cppLinkStdLib;
+
   CBuilder.library({
     required this.name,
     required this.assetId,
     this.sources = const [],
+    this.includes = const [],
     this.dartBuildFiles = const ['build.dart'],
     @visibleForTesting this.installName,
+    this.flags = const [],
     this.defines = const {},
     this.buildModeDefine = true,
     this.ndebugDefine = true,
     this.pic = true,
+    this.std,
+    this.cpp = false,
```

Comment:
Maybe this should be a language enum that defaults to `Language.c`. For example some compilers can also compile Objective-C, so making a `cpp` bool seems the wrong approach.